snuba/web/db_query.py

```diff
@@ -71,9 +72,17 @@ def encode_exception(self, value: SerializableException) -> bytes:
         return cast(str, rapidjson.dumps(value.to_dict())).encode("utf-8")
 
 
-cache: Cache[Result] = RedisCache(
-    redis_client, "snuba-query-cache:", ResultCacheCodec(), ThreadPoolExecutor()
-)
+DEFAULT_CACHE_PARTITION_ID = "default"
+
+cache_partitions: MutableMapping[str, Cache[Result]] = {
+    DEFAULT_CACHE_PARTITION_ID: RedisCache(
+        redis_client, "snuba-query-cache:", ResultCacheCodec(), ThreadPoolExecutor()
+    )
+}
```

Comment:
I am intentionally not initializing all cache partitions here and doing it lazily instead, because this module never has access to the cluster object it should run a query onto. Instead it has access to a reader so it needs to receive the partition_id from the Reader.
While it is true that the reader does not return a `partition_id` that is not configured in any cluster, nothing enforces that. So it is safer to initialize the partitions lazily when they are used.